scripts/mirna_extension.py

```diff
@@ -79,159 +123,132 @@ def extend_mirnas(
-                            mature_mir.start -= n
-                        else:
-                            mature_mir.start = 0
-
-                        if mature_mir.end + n < seq_lengths[seqid]:
-                            mature_mir.end += n
-                        else:
-                            mature_mir.end = seq_lengths[seqid]
+        assert isinstance(self.db, gffutils.FeatureDB)
 
-                        min_start = min(min_start, mature_mir.start)
-                        max_end = max(max_end, mature_mir.end)
+        feats_updated: list[gffutils.Feature] = []
 
-                        mirna.write(str(mature_mir) + "\n")
-
-                    start_diff = primary_mirna.start - min_start
-                    end_diff = max_end - primary_mirna.end
+        for pre in self.db.features_of_type(  # type: ignore
+            "miRNA_primary_transcript"
+        ):
+            feats_updated.extend(self.process_precursor(precursor=pre, n=n))
 
-                    primary_mirna.start = min_start
-                    primary_mirna.end = max_end
+        self.db_out = gffutils.create_db(
+            feats_updated,
+            dbfn=":memory:",
+            force=True,
+            keep_order=True,
+        )
 
-                    primary_mirna.attributes["Name"][0] += f"_-{start_diff}"
-                    primary_mirna.attributes["Name"][0] += f"_+{end_diff}"
+    def write_gff(self, path: Path, feature_type: Optional[str] = None) -> None:
+        """Write features to a GFF3 file.
 
-                primir.write(str(primary_mirna) + "\n")
+        Args:
+            path: Path to the output file.
+            feature_type: Feature type to write. If `None`, all features will be
+                written.
+        """
+        with open(path, "w", encoding="utf-8") as _file:
+            if not isinstance(self.db_out, gffutils.FeatureDB):
+                _file.write("")
+            elif feature_type is None:
+                for feature in self.db_out.all_features():  # type: ignore
+                    _file.write(str(feature) + "\n")
+            else:
+                for feature in self.db_out.features_of_type(  # type: ignore
+                    feature_type
+                ):
+                    _file.write(str(feature) + "\n")
 
 
 def parse_arguments():
-    """Command-line arguments parser."""
+    """Parse command-line arguments."""
+    description = """Extend miRNA start and end coordinates.
+
+Extends mature miRNA start and end coordinates in a GFF3 file by the indicated number of
+nucleotides, but not exceeding chromosome boundaries. If the extension causes the mature
+miRNA coordinates to exceed the boundaries of the corresponding precursor (see note in
+"Constraints" below), the precursor coordinates will be extended accordingly. In this
+case, the precursor name will be appended with "_-n" and/or "_+m", where n and m
+represent the extensions on the 5' and 3' end, respectively. The modified mature miRNA
```

Comment:
I don't think the "In this case, the precursor name will be appended with ..." statement is accurate. Regardless of the extension the precursor name be modified. If no extension is made, the name will end in "_-0_+0".

Reply:
Also, the whole description has no indentation.

Reply:
Good point about the description. If I recall correctly, I think the previous description also said or implied that names do not change if the boundaries are not shifted (but I may be wrong). In any case, what I described is _not_ what I implemented. 

Could you please update the description and/or code according to how you want/need it and make sure they match? 

There's no indentation because the triple quotes retain whitespace. So using indentation here would probably lead to random whitespace being included in the output, depending on how `argparse` formats and wraps the description. If it bothers you, you can construct the description from multiple parts instead, e.g., like so:

```py
description = (
    "This is a very long"
    " sentence, but we care"
    " about indentation."
)
```